### ISSUE TITLE
livenessprobe consumes too much memory 

### ISSUE THREAD

**Issue by leakingtapan** [kubernetes-csi/livenessprobe#45]
I ran the CSI driver with liveness probe for several days and the liveness probe memory usage turned to be 765MB which is abnormal compared to other csi sidecars. 

The follow is a screenshot from `htop`:

<img width="1280" alt="Screen Shot 2019-04-14 at 5 39 16 PM" src="https://user-images.githubusercontent.com/5259480/56101921-842f8180-5edd-11e9-96ac-527f202b136e.png">

Another side affect is my CSI driver turned to be crashing a lot more after liveness probe is enabled:

```
NAMESPACE     NAME                                                    READY   STATUS    RESTARTS   AGE
kube-system   dns-controller-64db5996cd-wvrmv                         1/1     Running   0          6d7h
kube-system   ebs-csi-controller-0                                    6/6     Running   36         5d18h
kube-system   ebs-csi-node-djvrx                                      3/3     Running   20         5d18h
kube-system   ebs-csi-node-g2c2k                                      3/3     Running   9          5d18h
```

Here is my liveness probe config:
```
          livenessProbe:
            httpGet:
              path: /healthz
              port: healthz
            initialDelaySeconds: 10
            timeoutSeconds: 3
            periodSeconds: 10
            failureThreshold: 5
```

**Comment by jsafrane** [comment]
Which version do you use? I checked both `-canary` and `v1.0.2` and it does not increase above ~15 MB after ~500 000 probe requests.

**Comment by leakingtapan** [comment]
It’s quay.io/k8scsi/livenessprobe:v1.0.1

**Comment by jsafrane** [comment]
It's indeed fixed in v1.0.2, see https://github.com/kubernetes-csi/livenessprobe/pull/27

/close

**Comment by yongzhang** [comment]
I have the same issue with v2.1.0
https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/341
